pkg/key/key.go

```diff
@@ -147,3 +149,21 @@ func GetBaseDomain(ctx context.Context, ctrlClient client.Client, clusterName, n
 
 	return baseDomain, nil
 }
+
+func GetAWSAccountID(awsClusterRoleIdentity *capa.AWSClusterRoleIdentity) (string, error) {
```

Comment:
Let's use opsctl's battle-tested function `extractAwsAccountIDfromARN` which uses the SDK for parsing